crates/adapterlib/src/transport.rs

```diff
@@ -269,17 +269,65 @@ pub enum NonFtInputReaderCommand {
     Transition(PipelineState),
 }
 
+pub struct InputQueueEntry<A> {
+    /// Data buffer to push to the circuit.
+    buffer: Option<Box<dyn InputBuffer>>,
+
+    /// Time when data in this buffer was received from the transport endpoint.
+    /// It is used to track the processing latency in different stages of the pipeline.
+    timestamp: DateTime<Utc>,
+
+    /// Start a transaction with the given label before pushing the buffer to the circuit
+    /// unless a transaction is already in progress.
```

Comment:
Not obvious to me why this is Option<Option<...>> (maybe it will become more obvious later)